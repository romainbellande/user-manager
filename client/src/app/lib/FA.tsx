import * as React from 'react';

export class FA extends React.Component<{size?: 2 | 3 | 5 | 7 }, {}> {
  public getSize(): string {
    return this.props.size ? ' fa-' + this.props.size + 'x' : '';
  }
  public render() {
    return (
      <i className={'fa fa-' + this.props.children + this.getSize()} />
    );
  }
}
