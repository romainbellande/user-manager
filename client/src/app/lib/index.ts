export * from './FA';
export * from './validators';
export * from './test-renderer';
