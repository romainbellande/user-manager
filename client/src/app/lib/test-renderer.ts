import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
const { HTMLAttributes, shallow, ShallowWrapper } = require('enzyme');

configure({ adapter: new Adapter() });

export const testRenderer = (element: any) => {
  return shallow(element, document.createElement('div'));
};
