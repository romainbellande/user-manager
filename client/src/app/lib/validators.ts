export class Validators {
  public static email(value: string) {
    return value && value.match(/^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/);
  }

  public static wordLength(value: string, minLength = 0, maxLength?: number) {
    return value && value.length > minLength && maxLength ? value.length < maxLength : true;
  }

  public static password(value: string) {
    // return value && value.match(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/);
    return value && value.match(/^.{8,}$/);
  }

  public static username(value: string) {
    return value && value.match(/[a-zA-Z]+/);
  }
}
