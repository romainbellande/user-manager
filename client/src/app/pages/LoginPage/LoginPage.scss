.LoginPage {
  // display: flex;
  // justify-content: center;
  // align-items: center;

  // &__form {
  //   width: 300px;
  // }
}
