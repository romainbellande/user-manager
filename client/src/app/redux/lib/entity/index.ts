export * from './entity-type-gen';
export * from './entity.reducer';
export * from './entity.state';
