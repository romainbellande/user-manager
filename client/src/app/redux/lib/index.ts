export * from './entity';
export * from './redux-actions';
export * from './redux-effect';
