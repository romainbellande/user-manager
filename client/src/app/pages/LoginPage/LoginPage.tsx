import './LoginPage.scss';

import * as React from 'react';
import { Form, FormApi, Text } from 'react-form';
import { Redirect, RouteComponentProps } from 'react-router-dom';
import { push } from 'react-router-redux';
const { connect } = require('react-redux');
const { withRouter } = require('react-router-dom');

import { IStore } from '../../redux';
import { Validators } from '../../lib';
import { AuthActions, IAuthState } from '../../redux/auth';
import { ICredentials, IUser } from '../../common';

export interface ILoginPage extends RouteComponentProps<any> {
  auth?: IAuthState;
  connection?: (credentials: ICredentials) => any;
}

@withRouter
@connect(
  (state) => ({ auth: state.auth }),
  (dispatch) => ({
    connection: (payload: ICredentials) => dispatch({ type: AuthActions.CONNECT, payload }),
  }),
)
export class LoginPage extends React.Component<ILoginPage, {}> {
  constructor(props, state) {
    super(props, state);
    this.onSubmit = this.onSubmit.bind(this);
  }
  public errorValidator(values: ICredentials): {} {
    return {
      email: !Validators.email(values.email) ? 'Please provide a valid email' : null,
      password: !Validators.password(values.password) ?
        'The password must contains minimum 8 characters' : null,
    };
  }

  public onSubmit(credentials: ICredentials, error, formApi: FormApi): void {
    this.props.connection(credentials);
  }
  public render() {
    return (
      <section className="LoginPage page">
        <Form
          validateError={this.errorValidator}
          onSubmit={this.onSubmit}
        >
          {this.renderForm}
        </Form>
        {this.redirect()}
      </section>
    );
  }

  public renderForm(formApi: FormApi): JSX.Element {
    return (
      <form className="LoginPage__form" onSubmit={formApi.submitForm}>
        <div className="form-group">
          <label htmlFor="login-email">Email</label>
          <Text id="login-email" type="email" className="form-control" field="email"/>
          <small className="form-text text-muted">We'll never share your email with anyone else.</small>
        </div>
        <div className="form-group">
          <label htmlFor="login-password">Password</label>
          <Text id="login-password" type="password" className="form-control" field="password" />
          <small className="form-text text-muted">We'll never share your email with anyone else.</small>
        </div>
        <button type="submit" className="btn btn-primary">Login</button>
      </form>
    );
  }

  public redirect() {
    return !!this.props.auth.token ? <Redirect to="/" /> : null;
  }
}
