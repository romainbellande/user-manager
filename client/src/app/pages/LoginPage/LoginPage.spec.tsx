import { LoginPage } from './LoginPage';

import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {BrowserRouter as Router, Route} from 'react-router-dom';

import { testRenderer } from '../../lib';

describe('LoginPage', () => {
  it('should render login page', () => {
    const div = document.createElement('div');
    // tslint:disable-next-line:no-angle-bracket-type-assertion
    const loginPage = testRenderer(
      <Router>
        <Route children={LoginPage} />
      </Router>);
    expect(loginPage).toMatchSnapshot();
    expect(loginPage.find('.LoginPage__form')).toBeDefined();
  });
});
