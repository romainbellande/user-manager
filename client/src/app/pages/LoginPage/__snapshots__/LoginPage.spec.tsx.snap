// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render login page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <Route>
      [Function]
    </Route>
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Route>
        [Function]
      </Route>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Route>
          [Function]
        </Route>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
